Automattic/jetpack
# Jetpack AI: add logo generator codebase to ai-client package

Closes #38422.

## Proposed changes:

* Copy logo generator code from Calypso's package to Jetpack's `ai-client` package
* Solve issues related to the moving, like broken dependencies, SCSS variables, backend requests that would not work on Jetpack sites, etc
* Add a raw entry point on the sidebar so we can trigger the modal and test it without doing the extension implementation yet

Note: this is the first pass of the moving process. I expect to do further changes on the code while we move the development forward. I also expect to have dirty things and leftovers on the commited code, that we want to clean up on the following PRs as we find them out. Some duplicate code is also present on the feature data reducer/selector machinery, that we may remove on the next steps.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable beta extensions on your testing site
* Go to the block editor
* Open the Jetpack sidebar, look for the AI Assistant section, then look for the "Generate Logo" button

<img width="250" alt="Screenshot 2024-07-19 at 07 27 20" src="https://github.com/user-attachments/assets/24e4844f-e64f-4de5-b152-79a24ee2d078">

* This is not the real implementation, is just a shortcut to trigger the generation modal for testing purposes
* Click the "Generate Logo" button
* Confirm the Logo Generator modal will show up, and that you get the first logo automatically

<img width="500" alt="Screenshot 2024-07-19 at 07 28 14" src="https://github.com/user-attachments/assets/e2c4aedf-71a0-4056-9849-f1f777591a55">

<img width="500" alt="Screenshot 2024-07-19 at 07 29 14" src="https://github.com/user-attachments/assets/259e2699-4a3b-45ce-981e-a93903a64598">

* Confirm you can keep generating logos on the modal
* Confirm the generated logos are saved to the media library
* Note: the "Use on Site" button is not working since the focus is to get the core functionality in place
* Important: repeat the test on a Simple site, to make sure the requests are running as expected
